mbf_mesh_nav/src/mesh_navigation_server.cpp

```diff
@@ -37,13 +37,94 @@
 
 #include "mbf_mesh_nav/mesh_navigation_server.h"
 
+#include <algorithm>
 #include <functional>
 
 #include <geometry_msgs/msg/pose_array.hpp>
 #include <mesh_map/mesh_map.h>
 #include <nav_msgs/msg/path.hpp>
 #include <rclcpp/logging.hpp>
 
+namespace {
+//! Helper function, intended for formatting available plugin types.
+//! Returns a string like this: [el1, el2, el3, ..., eln]
+std::string stringVectorToString(const std::vector<std::string>& vec) 
+{
+  std::stringstream s;
+  s << "[";
+  bool is_first = true;
+  for (const auto& item : vec)
+  {
+    s << (is_first ? "" : ", ") << item;
+    is_first = false;
+  }
+  s << "]";
+  return s.str();
+};
+
+/*!
+ * Helper function to reduce code duplication when loading plugins.
+ * Things can be either planners, controllers or recovery behaviors.
+ * @tparam AbstractThing Abstract type of the thing to load. Needs to be the base class of SimpleThing and MeshThing. E.g. mbf_abstract_core::AbstractPlanner.
+ * @tparam SimpleThing Simple type of the thing to load. E.g. mbf_simple_core::SimplePlanner
+ * @tparam MeshThing Mesh type of the thing to load. E.g. mbf_simple_core::MeshPlanner
+ * @param thing_type typename of the thing that shall be loaded, usually configured by user via ros params.
+ * @param which_thing should be "planner", "controller", or "recovery behavior", used only for logging.
```

Comment:
You did not document the logger param. Was that intentional?

Reply:
Yes, I tend to be somewhat lazy with documentation, especially for internal functions where the context not broad.
I only added some docs for parameters that seemed unclear to me. Mainly for explaining the weird `thing` stuff, because I couldn't come up with a better name.
I think my documentation for the logger would be superfluous (i.e. "it's a logger, it is used for logging", okay cpt obvious).